adapter/PreciceInterface.c

```diff
@@ -420,6 +420,7 @@ void Precice_FreeData( SimulationData * sim )
 		if( sim->preciceInterfaces[i] != NULL ){
 			free( sim->preciceInterfaces[i] );
 		}
```

Comment:
As you are already cleaning this up:

```suggestion
		free( sim->preciceInterfaces[i] );
```

`free(NULL);` does nothing by definition.